indico/modules/events/registration/controllers/display.py

```diff
@@ -258,8 +258,12 @@ def _checkParams(self, params):
     def _process(self):
         form = make_registration_form(self.regform)()
         if form.validate_on_submit():
-            registration = create_registration(self.regform, form.data, self.invitation)
-            return redirect(url_for('.display_regform', registration.locator.registrant))
+            if self.regform.limit_reached:
```

Comment:
I would move this check to the very beginning (could even be in `_checkProtection`). Even displaying the form doesn't make much sense once the limit has been reached.